File: branches/with-libpq-backend/src-testing/dbModel/Schema.cpp
#include "Schema.h"

#include "Database.h"
#include "Sequence.h"
#include "Table.h"

Schema::Schema(Database* db, const QString& name)
  : DbInterfacePart(db, name)
{}

Schema::~Schema()
{}

Sequence* Schema::createSequence(const QString& name)
{
  _sequences.append(new Sequence(this, name));
  return _sequences.last();
}

QList<Sequence*>::iterator Schema::getFirstSequence() 
{
  return _sequences.begin();
}

QList<Sequence*>::iterator Schema::getLastSequence()
{
  return _sequences.end();
}

Table* Schema::createTable(const QString& name)
{
  _tables.append(new Table(this, name));
  return _tables.last();
}

QList<Table*>::iterator Schema::getFirstTable() 
{
  return _tables.begin();
}

QList<Table*>::iterator Schema::getLastTable()
{
  return _tables.end();
}

File: tools/CodeGenerators/codegen/src/Schema.py
from Entity import *
from Sequence import *
from Table import *

class Schema(Entity):
    def __init__(self, database, name):
        Entity.__init__(self, name)
        self.database = database
        self.sequences = {}
        self.tables = {}
    def createSequence(self, name):
        self.sequences[name] = Sequence(self, name)
        return self.sequence(name)
    def sequence(self, name):
        return self.sequences[name]
    def createTable(self, name,  
                    hasIdColumn=True, 
                    hasNameColumn=False, nameColumnIsUnique=False, 
                    hasDescriptionColumn=False):
        self.tables[name] = Table(self, name, 
                                  hasIdColumn=hasIdColumn, 
                                  hasNameColumn=hasNameColumn, 
                                  nameColumnIsUnique=nameColumnIsUnique, 
                                  hasDescriptionColumn=hasDescriptionColumn)
        return self.table(name)
    def table(self, name):
        return self.tables[name]
